common/js/src/popup-window/popup-opener.js

```diff
@@ -147,50 +145,35 @@ GSC.PopupOpener.createWindow = function(url, windowOptions, opt_data) {
  * @param {string} url
  * @param {!WindowOptions=} opt_windowOptions
  * @param {!Object=} opt_data Optional data to be passed to the created dialog.
- * @return {!goog.Thenable}
+ * @return {!Promise}
  */
 GSC.PopupOpener.runModalDialog = function(url, opt_windowOptions, opt_data) {
   const createWindowOptions =
       goog.object.clone(DEFAULT_DIALOG_CREATE_WINDOW_OPTIONS);
   if (opt_windowOptions !== undefined)
     Object.assign(createWindowOptions, opt_windowOptions);
 
-  if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION)
-    lastUsedPopupId++;
-
-  const promiseResolver = goog.Promise.withResolver();
-
-  // Set data for corresponnding SCC mode
-  const modifiedData =
-      (GSC.Packaging.MODE === GSC.Packaging.Mode.APP ?
-           {
-             'resolveModalDialog': promiseResolver.resolve,
-             'rejectModalDialog': promiseResolver.reject
-           } :
-           {'popup_id': lastUsedPopupId});
-
-  // Set promiseResolver for SCC extension mode
-  if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION) {
-    goog.global[`googleSmartCard_resolveModalDialog${lastUsedPopupId}`] =
-        promiseResolver.resolve;
-    goog.global[`googleSmartCard_rejectModalDialog${lastUsedPopupId}`] =
-        promiseResolver.reject;
-  }
-
+  // Additionally pass the auto-generated popup ID. The popup will use it to
+  // talk back to us when sending the result.
+  ++lastUsedPopupId;
```

Comment:
Nit: Isn't `lastUsedPopupId++;` more readable? I also don't see any difference in functionality.

Reply:
done (I'm personally a bit more inclined to the preincrement version by default as it has simpler semantics than postincrement, but there's obviously no strong argument for/against in simple cases like this)